### PR TITLE
rename `react-addons-dom-factories` to `react-dom-factories`

### PR DESCRIPTION
**Pull request by flarnie**
**what is the change?:**
`grep -r "react-addons-dom-factories" ./src` and the same in `./packages`
then `%s /react-addons-dom-factories/react-dom-factories/gc`

**why make this change?:**
React dom factories were never part of the 'addons' and we want to minimize the
whole 'addons' thing, since we are deprecating the `React.addons.*` API.

**test plan:**
`yarn test`
and manually look at the README that was updated.

**issue:**
https://github.com/facebook/react/issues/9398

### CODE REVIEW

**Comment by flarnie** [comment]
Note to self: cherry-pick onto master

**Comment by gaearon** [comment]
>Note to self: cherry-pick onto master

Do we need to? I thought this package won't receive more updates.

File: packages/react-dom-factories/README.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 pre-configured DOM factories. These predefined factories have been
-moved to the `react-addons-dom-factories` library.
+moved to the `react-dom-factories` library.
 
 ## Example
 
 ```javascript
 import ReactDOM from 'react-dom';
-import DOM from 'react-addons-dom-factories'; // ES6
+import DOM from 'react-dom-factories'; // ES6
```
Comment (gaearon) [inline packages/react-dom-factories/README.md]: We can either remove `// ES6` here or provide an ES5 example.

**Comment by flarnie** [comment]
@spicyj
> The build scripts are already configured correctly?
I don't know. Let's figure it out. :)

@gaearon
> Do we need to? I thought this package won't receive more updates.
You are correct.
I was thinking that updating the package `README` would be good, in case it confuses people to see the wrong name there. https://github.com/facebook/react/tree/master/packages/react-dom-factories
Hopefully we'll have dropped support and won't be warning for this at all in 16.0, but just on the off-change that we delay actually removing some of these APIs I'd like the warnings to be the same as in 15.6.

File: packages/react-dom-factories/README.md (hunk 1 above)
Comment (flarnie) [inline packages/react-dom-factories/README.md]: oops - just noticed this. will fix.

**Comment by gaearon** [comment]
Ah okay. I thought we can just delete that folder for 16.0.

**Comment by NE-SmallTown** [comment]
I notice that the both `react-addons-dom-factories` and `react-dom-factories` are not published in npm,but the React 16 warning it now,so I want to know is there any suggestion to solve this?Thanks.

**Comment by gaearon** [comment]
Not sure what you mean. Afaik this warning hasn't made it into any release.

**Comment by flarnie** [comment]
@NE-SmallTown might have seen that the warning was merged into master as well as 15.6, in https://github.com/facebook/react/pull/8356 so they are right that it's present in the latest **unreleased** 16.0-alpha.

afaik we have not done an alpha-release of the 16 branch yet that includes that warning. So @gaearon is also correct.

Instead of cherry-picking this onto master, we should do a commit that removes it and the warning both on the 16.0 branch. I'll open an issue to track that. (later this weekend probably)

 Also I expect to publish the `react-dom-factories` package soon, hopefully that mitigates the concern.

**Comment by NE-SmallTown** [comment]
Yea,I am using react 16 now.But you say 16 not includes this warning,I think this is not true because I get this warning.



And I find the src code of this in [React.js](https://github.com/facebook/react/blob/964c263d8fc6e7119f5773f7fa857c9985db40cf/src/isomorphic/React.js#L160) file

**Comment by flarnie** [comment]
I believe you @NE-SmallTown - now I'm curious how it's happening.

Which of the 16 alpha releases are you using? Or are you using a build directly from the react master branch? If you have pulled the recent React code directly from master any time after Apr. 24th then you would indeed get the inactionable warning. https://github.com/facebook/react/pull/8356

I think the latest alpha-release of 16.0 is https://github.com/facebook/react/releases/tag/v16.0.0-alpha.3 and when I check the source code I don't see it there either.

We added this warning on

**Comment by gaearon** [comment]
Alpha 3 is definitely not the latest, we released at least alpha 8 (and maybe more). I don't think we documented them anywhere but you can check what `react@next` resolves to.

Generally alphas may contain unactionable warnings etc. We don't recommend anyone to use React alphas outside of React Native.

**Comment by NE-SmallTown** [comment]
> Which of the 16 alpha releases are you using?

I am using alpha-12.0,I don't specify concrete version,I just use `npm install react@next`,and then the version is alpha-12.0(IIRC one or two weeks ago it is 11.0).

> Or are you using a build directly from the react master branch?

No,only when I learn fiber in the debugger-fiber directory I will do this.

>  Generally alphas may contain unactionable warnings etc. We don't recommend anyone to use React alphas outside of React Native.

Yea,don't worry,please trust that I am not blaming or complaining anyone or anything.I just want to experience fiber and some new features in advance.

**Pull request merged**